spark/src/main/scala/org/apache/spark/sql/delta/abstractActions/AbstractProtocol.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.abstractActions
+
+/**
+ * Abstract trait for protocol actions in Delta. This trait provides a common
+ * abstraction that can be implemented by both Spark's V1 Protocol and Kernel's Protocol
+ * in V2 connector. The V2 connector will implement adapters for reusing V1 utilities.
+ */
+trait AbstractProtocol {
```

Comment:
these are mainly intended for interop between v1 and v2 ... right? can we just put them under directory named v2/interop/

with a readme in the dir explaining the purpose of these classes?

Reply:
Yes, and update the pr accordingly